packages/big-design/src/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -0,0 +1,71 @@
+import React, { ButtonHTMLAttributes, Fragment, HTMLAttributes, memo, ReactNode, useMemo } from 'react';
+
+import { MarginProps } from '../../mixins';
+import { Dropdown, DropdownProps } from '../Dropdown';
+import { Flex } from '../Flex';
+
+import { StyledButton } from './styled';
+
+export interface ButtonGroupItem extends ButtonHTMLAttributes<HTMLButtonElement> {
+  content: string | ReactNode;
+}
+
+export interface ButtonGroupDropdownItem extends Omit<DropdownProps, 'toggle'> {
+  toggle: ButtonGroupItem;
+}
+
+export interface ButtonGroupProps extends HTMLAttributes<HTMLDivElement>, MarginProps {
+  items: Array<ButtonGroupItem | ButtonGroupDropdownItem>;
```

Comment:
So I think we should make this API a little more consistent with some of our existing patterns, like the `Modal` `actions` prop: https://github.com/bigcommerce/big-design/blob/master/packages/big-design/src/components/Modal/Modal.tsx#L22.

I'll talk a little be more about my vision of `ButtonGroupDropdownItem` in a new comment.